packages/dashboard-core-plugins/src/panels/IrisGridPanel.tsx

```diff
@@ -151,11 +159,74 @@ export interface IrisGridPanelProps {
   getDownloadWorker: () => Promise<ServiceWorker>;
 
   // Load a plugin defined by the table
-  loadPlugin: (pluginName: string) => Plugin;
+  loadPlugin: (pluginName: string) => TablePlugin;
 
   theme: IrisGridThemeType;
 }
 
+export interface TablePluginElement {
+  getMenu?: (data: ContextMenuData) => ResolvableContextAction[];
+}
+
+export interface TablePluginProps {
+  /**
+   * Apply filters to the table
+   * @param filters Filters to apply to the table
+   */
+  filter: (filters: InputFilter[]) => void;
+
+  /** @deprecated Use `filter` instead */
+  onFilter: (filters: InputFilter[]) => void;
+
+  /**
+   * Set columns that should always be fetched, even if they're outside the viewport
+   * @param pluginFetchColumns Names of columns to always fetch
+   */
+  fetchColumns: (pluginFetchColumns: ColumnName[]) => void;
+
+  /** @deprecated Use `fetchColumns` instead */
+  onFetchColumns: (pluginFetchColumns: ColumnName[]) => void;
+
+  /**
+   * The table this plugin was associated with
+   */
+  table: Table;
+
+  /**
+   * The IrisGridPanel displaying this table
+   */
+  panel: IrisGridPanel;
+
+  /** @deprecated Import components from @deephaven/components and @deephaven/iris-grid packages instead */
+  components: typeof PLUGIN_COMPONENTS;
+
+  /**
+   * Current user information
+   */
+  user: User;
+
+  /**
+   * Current user workspace data
+   */
+  workspace: Workspace;
```

Comment:
Do we want to provide these to the plugin or should it be up to the plugin to fetch these from a context? I guess the workspace is in redux, but the user has a context

Reply:
Yea it's already passed in as a prop. Makes sense to deprecate and recommend just pulling it from redux though... and since it wasn't _really_ documented before anyway, I'll just not doc it.